src/topology/paracompact.lean

```diff
@@ -0,0 +1,231 @@
+  -- `Kdiff (n + 1) = K (n + 2) \ interior (K (n + 1))` such that
+  -- `B (c n i) (r n i) ⊆ interior (K (n + 3)) \ K n`
+  have : ∀ n (x ∈ Kdiff (n + 1)), (K n)ᶜ ∈ 𝓝 x,
+    from λ n x hx, mem_nhds_sets (K.is_closed n) (λ hx', hx.2 $ K.subset_interior_succ _ hx'),
+  choose! r hrp hr using (λ n x hx, (hB x).mem_iff.1 (inter_mem_sets (hs x) (this n x hx))),
+  have hxr : ∀ n (x ∈ Kdiff (n + 1)), B x (r n x) ∈ 𝓝 x,
+    from λ n x hx, (hB x).mem_of_mem (hrp _ _ hx),
+  choose T hTK hT using λ n, (Kdiffc (n + 1)).elim_nhds_subcover _ (hxr n),
+  -- Finally, we take the union of all these coverings
+  refine ⟨Σ n, ↥(T n : set X), λ a, a.2, λ a, r a.1 a.2, _, _, _, _⟩,
+  { rintro ⟨n, x, hx⟩, exact hrp _ _ (hTK _ _ hx) },
+  { refine Union_eq_univ_iff.2 (λ x, _),
+    obtain ⟨c, hcT, hcx⟩ : ∃ c ∈ T (K'.find x), x ∈ B c (r (K'.find x) c) :=
+      mem_bUnion_iff.1 (hT _ (hKcov x)),
+    exact ⟨⟨_,  c, hcT⟩, hcx⟩ },
+  { intro x,
+    refine ⟨interior (K (K'.find x + 3)),
+      mem_nhds_sets is_open_interior (K.subset_interior_succ _ (hKcov x).1), _⟩,
+    have : (⋃ k ≤ K'.find x + 2, (range $ sigma.mk k) : set (Σ n, ↥(T n : set X))).finite,
+      from (finite_le_nat _).bUnion (λ k hk, finite_range _),
+    apply this.subset, rintro ⟨k, c, hc⟩,
+    simp only [mem_Union, mem_set_of_eq, mem_image_eq, subtype.coe_mk],
+    rintro ⟨x, hxB : x ∈ B c (r k c), hxK⟩,
+    refine ⟨k, _, ⟨c, hc⟩, rfl⟩,
+    have := (mem_compl_iff _ _).1 (hr k c (hTK _ _ hc) hxB).2,
+    contrapose! this with hnk,
+    exact K.subset hnk (interior_subset hxK) },
+  { rintro ⟨n, x, hx⟩,
+    exact subset.trans (hr n x $ hTK _ _ hx) (inter_subset_left _ _) }
+end
+
+/-- A locally compact sigma compact Hausdorff space is paracompact. See also
+`refinement_of_locally_compact_sigma_compact_of_nhds_basis` for a more precise statement. -/
+@[priority 100] -- See note [lower instance priority]
+instance paracompact_of_locally_compact_sigma_compact [locally_compact_space X]
+  [sigma_compact_space X] [t2_space X] : paracompact_space X :=
+begin
+  refine ⟨λ α s ho hc, _⟩,
+  choose i hi using Union_eq_univ_iff.1 hc,
+  rcases refinement_of_locally_compact_sigma_compact_of_nhds_basis nhds_basis_opens
+    (s ∘ i) (λ x, mem_nhds_sets (ho _ ) (hi _)) with ⟨β, c, t, hto, htc, htf, hsub⟩,
+  exact ⟨β, t, λ x, (hto x).2, htc, htf, λ b, ⟨i $ c b, hsub _⟩⟩
+end
+
+/- Dieudonné‘s theorem: a paracompact Hausdorff space is normal. Formalization is based on the proof
+at [ncatlab](https://ncatlab.org/nlab/show/paracompact+Hausdorff+spaces+are+normal). -/
+lemma normal_of_paracompact_t2 [t2_space X] [paracompact_space X] : normal_space X :=
+begin
+  /- It suffices to learn how to go from points to a set on one side. Then we can apply
+  this procedure to one set, then to the other set. -/
+  suffices : ∀ (s t : set X), is_closed s → is_closed t →
+    (∀ x ∈ s, ∃ u v, is_open u ∧ is_open v ∧ x ∈ u ∧ t ⊆ v ∧ disjoint u v) →
+    ∃ u v, is_open u ∧ is_open v ∧ s ⊆ u ∧ t ⊆ v ∧ disjoint u v,
```

Comment:
Since the proof of the lemma is roughly the same size as the conclusion, can you exchange the two to do it in the direct order? I.e., replace `suffices :` with `have :`, give the proof, and then conclude.